src/Flow.jl

```diff
@@ -137,19 +143,21 @@ end
 Integrate the `Flow` one time step using the [Boundary Data Immersion Method](https://eprints.soton.ac.uk/369635/)
 and the `AbstractPoisson` pressure solver to project the velocity onto an incompressible flow.
 """
-@fastmath function mom_step!(a::Flow,b::AbstractPoisson)
+@fastmath function mom_step!(a::Flow{N},b::AbstractPoisson) where N
     a.u⁰ .= a.u; scale_u!(a,0)
     # predictor u → u'
+    U = BCTuple(a.U,time(a),N)
     conv_diff!(a.f,a.u⁰,a.σ,ν=a.ν,perdir=a.perdir)
-    accelerate!(a.f,time(a),a.g)
-    BDIM!(a); BC!(a.u,a.U,a.exitBC,a.perdir)
-    a.exitBC && exitBC!(a.u,a.u⁰,a.U,a.Δt[end]) # convective exit
-    project!(a,b); BC!(a.u,a.U,a.exitBC,a.perdir)
+    accelerate!(a.f,time(a),a.g,a.U)
```

Comment:
is the last argument supposed to be `a.U` or `U`?

Reply:
Yes, this is correct. It sets the correct type dispatch for `accelerate`